### PR TITLE
Card scripts for Lantern of Revealing and Marching Duodrone

### LINKED ISSUE #888: Lantern of Revealing

### PR DESCRIPTION
**Pull request by xaviermd**
Closes #888 and #893

### CODE REVIEW

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/lantern_of_revealing.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+A:AB$ Dig | Cost$ 4 T | DigNum$ 1 | ChangeNum$ All | ForceRevealToController$ True | Optional$ True | PromptToSkipOptionalAbility$ True | OptionalAbilityPrompt$ Would you like to put the land onto the battlefield tapped? | ChangeValid$ Land | DestinationZone$ Battlefield | Tapped$ True | DestinationZone2$ Library | LibraryPosition2$ 0 | RememberChanged$ True | SubAbility$ DBMoveToBottom | SpellDescription$ Look at the top card of your library. If it’s a land card, you may put it onto the battlefield tapped. If you don’t put the card onto the battlefield, you may put it on the bottom of your library.
+SVar:DBMoveToBottom:DB$ Dig | DigNum$ 1 | DestinationZone$ Library | Optional$ True | LibraryPosition$ -1 | LibraryPosition2$ 0 | ConditionPresent$ Card | ConditionDefined$ Remembered | ConditionCompare$ EQ0 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+Oracle:{T}: Add one mana of any color.\n{4}, {T}: Look at the top card of your library. If it’s a land card, you may put it onto the battlefield tapped. If you don’t put the card onto the battlefield, you may put it on the bottom of your library.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/lantern_of_revealing.txt:8]: it’s = it's
don’t = don't

File: forge-gui/res/cardsfolder/upcoming/lantern_of_revealing.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Name:Lantern of Revealing
+ManaCost:3
+Types:Artifact
+A:AB$ Mana | Cost$ T | Produced$ Any | SpellDescription$ Add one mana of any color.
+A:AB$ Dig | Cost$ 4 T | DigNum$ 1 | ChangeNum$ All | ForceRevealToController$ True | Optional$ True | PromptToSkipOptionalAbility$ True | OptionalAbilityPrompt$ Would you like to put the land onto the battlefield tapped? | ChangeValid$ Land | DestinationZone$ Battlefield | Tapped$ True | DestinationZone2$ Library | LibraryPosition2$ 0 | RememberChanged$ True | SubAbility$ DBMoveToBottom | SpellDescription$ Look at the top card of your library. If it’s a land card, you may put it onto the battlefield tapped. If you don’t put the card onto the battlefield, you may put it on the bottom of your library.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/lantern_of_revealing.txt:5]: it’s = it's
don’t = don't

(change the curly stuff)

File: forge-gui/res/cardsfolder/upcoming/marching_duodrone.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Name:Marching Duodrone
+ManaCost:2
+Types:Artifact Creature Construct
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ DBTreasure | TriggerDescription$ Whenever CARDNAME attacks, each player creates a Treasure token.
+SVar:DBTreasure:DB$ Token | TokenAmount$ 1 | TokenScript$ c_a_treasure_sac | TokenOwner$ Player
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/marching_duodrone.txt:6]: DeckHas:Ability$Token|Sacrifice & Type$Artifact|Treasure

File: forge-gui/res/cardsfolder/upcoming/marching_duodrone.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Name:Marching Duodrone
+ManaCost:2
+Types:Artifact Creature Construct
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ DBTreasure | TriggerDescription$ Whenever CARDNAME attacks, each player creates a Treasure token.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/marching_duodrone.txt:5]: TriggerDescription$ Whenever CARDNAME attacks, each player creates a Treasure token. (It's an artifact with "{T}, Sacrifice this artifact: Add one mana of any color.")

**Pull request merged**